### DISCUSSION TITLE
Automated filesystem scanning

### DISCUSSION THREAD

**Discussion by depau** [photoprism/photoprism#3980] [Q&A] [answered]
Hi! I use Nextcloud for photo storage and sync, and I have Photoprism configured to use the Nextcloud sync directory as the originals directory.

It's a bit annoying that I have to trigger a manual library rescan or else Photoprism won't be updated. Is there a way to enable automatic filesystem changes detection, i.e. via inotify?

**Comment by graciousgrey** [comment, marked as answer]
Hello,

At the moment, [automatic indexing](https://docs.photoprism.app/user-guide/library/originals/#automatic-indexing) is only triggered when files are added to the originals folder via webDAV. Alternatively, you can use a scheduler or cron job to perform indexing e.g. once per night. 

https://docs.photoprism.app/getting-started/docker-compose/#introduction
https://dl.photoprism.app/docker/scheduler/

A general file system watcher is planned, but we did not get to it yet.

- https://github.com/photoprism/photoprism/issues/2608

**Reply by depau** [reply]
Thanks! I set up a scheduled job using systemd to run indexing every 5m unless the previous indexing hasn't finished yet.

**Reply by lastzero** [reply]
Note that it is now also possible to automatically rescan your library at regular intervals by setting `PHOTOPRISM_INDEX_SCHEDULE` to e.g. `"@every 3h"` in your global configuration (scans are skipped if another indexing or import job is already running):
- https://docs.photoprism.app/getting-started/config-options/#indexing
